google-account-plugin/src/com/google/gct/login/ui/GoogleLoginActionButton.java

```diff
@@ -58,8 +59,22 @@ public void updateUi() {
       if(image == null) {
         myPresentation.setIcon(GoogleLoginIcons.DEFAULT_USER_AVATAR);
       } else {
-        Image scaledImage = image.getScaledInstance(16, 16, Image.SCALE_SMOOTH);
-        myPresentation.setIcon(new ImageIcon(scaledImage));
+        int side = 16;
+        Image scaledImage = image.getScaledInstance(side, side, Image.SCALE_SMOOTH);
+        ImageIcon icon = new ImageIcon(scaledImage);
+
+        MediaTracker tracker = new MediaTracker(this);
+        tracker.addImage(image, 0);
+        tracker.addImage(scaledImage, 1, side, side);
+        // Wait until images load
+        try {
+          tracker.waitForAll(5000);
```

Comment:
+1

On Tue, Jan 12, 2016 at 9:46 AM, Elliotte Rusty Harold <
notifications@github.com> wrote:

> In
> google-account-plugin/src/com/google/gct/login/ui/GoogleLoginActionButton.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/336#discussion_r49462121>
> :
>
> > @@ -58,8 +59,22 @@ public void updateUi() {
> >        if(image == null) {
> >          myPresentation.setIcon(GoogleLoginIcons.DEFAULT_USER_AVATAR);
> >        } else {
> > -        Image scaledImage = image.getScaledInstance(16, 16, Image.SCALE_SMOOTH);
> > -        myPresentation.setIcon(new ImageIcon(scaledImage));
> > +        int side = 16;
> > +        Image scaledImage = image.getScaledInstance(side, side, Image.SCALE_SMOOTH);
> > +        ImageIcon icon = new ImageIcon(scaledImage);
> > +
> > +        MediaTracker tracker = new MediaTracker(this);
> > +        tracker.addImage(image, 0);
> > +        tracker.addImage(scaledImage, 1, side, side);
> > +        // Wait until images load
> > +        try {
> > +          tracker.waitForAll(5000);
>
> Looking into it, yes, potenitally it does look like it might get executed
> on the UI thread because updateUI can be called via this method:
>
> @Override
> public void notifyStatusIndicator() {
>   if (myButton != null) {
>     ApplicationManager.getApplication().invokeLater(new Runnable() {
>       @Override
>       public void run() {
>         myButton.updateUi();
>       }
>     });
>   }
> }
>
> We shoudl probably just table this unless the bug resurfaces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/336/files#r49462121>
> .
>